pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -329,18 +342,26 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 
 	before := tr.Status.GetCondition(apis.ConditionSucceeded)
 
-	updateStatusFromPod(tr, pod, c.resourceLister, c.KubeClientSet, c.Logger)
+	addReady := updateStatusFromPod(tr, pod, c.resourceLister, c.KubeClientSet, c.Logger)
 
 	after := tr.Status.GetCondition(apis.ConditionSucceeded)
 
+	if addReady {
+		if err := c.updateReady(pod); err != nil {
+			return err
+		}
+	}
+
 	reconciler.EmitEvent(c.Recorder, before, after, tr)
 
 	c.Logger.Infof("Successfully reconciled taskrun %s/%s with status: %#v", tr.Name, tr.Namespace, after)
 
 	return nil
 }
 
-func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLister listers.PipelineResourceLister, kubeclient kubernetes.Interface, logger *zap.SugaredLogger) {
+// updateStatusFromPod modifies the task run status based on the pod and then returns true if the pod is running and
+// all sidecars are ready
+func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLister listers.PipelineResourceLister, kubeclient kubernetes.Interface, logger *zap.SugaredLogger) bool {
```

Comment:
especially now that this function is changing substantially, can it be moved out into a different package and have some unit tests added? (that might be a lot to do all at once but we could start with just a couple simple test - it would be more than we have currently!)

it would probably also mean you'd have to break this function up a bit - which I think would be good to do cuz it's getting a bit complex

Reply:
I've moved it out along with its associated unit tests into a new `taskrun/status` package.  I've also moved all of the `reason...` variables into constants inside this package as well.  A couple of other functions needed to move along with it.  The diff's quite a bit bigger now but it felt good to refactor it.

Reply:
awesome!!

quick heads up that it looks like the `status` package will be added in https://github.com/tektoncd/pipeline/pull/963 as well (more evidence that it's a good name imo!) so there might be some conflicts for someone